### PR TITLE
Add Coralogix Exporter First PR

### PR DESCRIPTION
**Pull request by ofirshmuel**
**Description:** <Describe what has changed.>

This PR adds a new Coralogix exporter, the exporter sends traces to Coralogix.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.
All tests file in "coralogixexporter" folder

**Documentation:** <Describe the documentation added.>
README and configuration file describe how to use this exporter

### CODE REVIEW

File: exporter/coralogixexporter/README.md
Comment (mx-psi) [inline exporter/coralogixexporter/README.md:24]: I assume this functionality (and the rest of environment variables) will be added on a future PR?

File: exporter/coralogixexporter/README.md
Comment (mx-psi) [inline exporter/coralogixexporter/README.md:2]: You may want to add a comment saying that this component is in development

File: internal/components/components.go
Comment (mx-psi) [inline internal/components/components.go:218]: Adding the exporter to `internal/components` should be done on the last PR per the contribution guidelines. The main `go.mod` should have no changes either

File: exporter/coralogixexporter/config_test.go
Comment (mx-psi) [inline exporter/coralogixexporter/config_test.go:32]: This code should either be removed or uncommented. I think this PR is a good place to add tests for configuration loading, and `TestExporter` can be added later

File: exporter/coralogixexporter/factory.go
Comment (mx-psi) [inline exporter/coralogixexporter/factory.go:49]: The Collector already logs when a component is created or initialized, so this log is not necessary

File: exporter/coralogixexporter/factory.go
Comment (mx-psi) [inline exporter/coralogixexporter/factory.go:50]:
```suggestion
	return nil, fmt.Errorf("Coralogix exporter is not implemented yet")
```

File: exporter/coralogixexporter/factory.go
Comment (mx-psi) [inline exporter/coralogixexporter/factory.go:51]: You should remove this commented code

**Review by mx-psi — CHANGES_REQUESTED** [review]
Thanks for splitting the original PR. I left some comments, @oded-dd can you review too?

File: exporter/coralogixexporter/README.md
Comment (ofirshmuel) [reply exporter/coralogixexporter/README.md:24]: Yes, this functionality will be added on a future PR.

File: go.mod
Comment (mx-psi) [inline go.mod:483]: go.mod file should not be modified in this PR (that's why CI is failing)

File: exporter/coralogixexporter/go.mod
Comment (mx-psi) [inline exporter/coralogixexporter/go.mod:7]: This was bumped to https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/42a06f91316730473b00cee4284acaddd88bd3ff/go.mod#L111 can you update here?

**Review by mx-psi — COMMENTED** [review]
Looking good, just a couple more things from my side. Also @oded-dd can you review?

File: exporter/coralogixexporter/go.mod
Comment (ofirshmuel) [reply exporter/coralogixexporter/go.mod:7]: Yes, I will update the changes

**Review by mx-psi — APPROVED** [review]
Looks good from my side if tests pass, I will mark as `ready-to-merge` once we have a review from the other CODEOWNER

**Review by oded-dd — APPROVED** [review]

**Review by mx-psi — COMMENTED** [review]
Looks like there is one more thing to be taken care of (hopefully the last!)

**Comment by jpkrohling** [comment]
Given this week's SIG meeting discussion about new components, should we require a sponsor for this?

**Comment by bogdandrutu** [comment]
@jpkrohling what is Coralogix? Sorry for my ignorance ...

**Comment by oded-dd** [comment]
@bogdandrutu Coralogix is a real-time insights and trend analysis for logs, metrics, tracing (soon once approved :)) and security data with no reliance on storage or indexing

**Comment by oded-dd** [comment]
@mx-psi any update on this PR? Can it be merged?

**Comment by mx-psi** [comment]
@oded-dd @ofirshmuel It looks like part of the lint was not executing before and is now failing because you are missing the [last point in this list](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-new-components) about adding the component to `versions.yaml`.

There is also the sponsoring question raised by @bogdandrutu and @jpkrohling above (but that's a matter to solve among the Collector contrib approvers & maintainers, so it won't block approval)

**Comment by ofirshmuel** [comment]
@mx-psi thanks, I fixed version.yaml, can it be merged?

**Review by mx-psi — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]
We haven't defined how the sponsoring process looks like in its entirety yet. What we agreed was that there should be a sponsor, ideally a volunteer one. If none can be found, we'll round-robin among approvers.

I'm OK with merging this without sorting out the sponsor, as we did define that we will always accept vendor-specific components.

**Pull request merged**
